### PR TITLE
Observe ngShow and ngHide directives

### PR DESCRIPTION
**Pull request by willtj**

### CODE REVIEW

**Comment by solidspark** [comment]
This currently fails CI:

```
.........
PhantomJS 1.9 uiSelect2 with a <select> element should observe the multiple attribute FAILED
    Expected false to be true.
............................
Firefox 19.0 uiSelect2 with a <select> element should observe the multiple attribute FAILED
    Expected false to be true.
    @/home/travis/build/angular-ui/ui-select2/test/select2Spec.js:176
```

Can you take a look and make sure you didn't break anything in regards to `multiple`?

File: src/select2.js [hunk 1]
```diff
@@ -174,9 +174,35 @@ angular.module('ui.select2', []).value('uiSelect2Config', {}).directive('uiSelec
...
+            }
+            else if ( newVal === false ){
+              $timeout(init);
+            }
+          });
+        }
+        
+        var init = function () {
+          // Check if the select should be hidden
+          if ( ( attrs.ngShow && scope.$eval(attrs.ngShow) === false ) || ( attrs.ngHide && scope.$eval(attrs.ngHide) === true ) ){
```
Comment (ProLoser) [inline src/select2.js:202]: This check seems absolutely silly. If you're using CSS to hide something, you should NOT be preventing the plugin from initializing.

**Comment by willtj** [comment]
The CI errors seem to be unrelated from what I can tell - a clean repo still fails those tests for me.

File: src/select2.js (hunk 1 above)
Comment (willtj) [inline src/select2.js:202]: I agree it doesn't look sensible. I'm using an Angular expression to determine whether to show or hide the `<select>` element, and currently when the element is hidden, select2 still renders. The better approach would probably be to set `display:none` on the select2 div when it's set on the attached `select`, but I couldn't see a way of doing that. Any suggestions?

Comment (ProLoser) [inline src/select2.js:202]: Frankly, I don't like explicitly adding support for each. and. every. angular. directive. I'd wrather someone wrap the `<select>` inside another DOM element and put the hide/show on THAT instead. No hassle, no work, no patches. The bottom line is that this implementation of select2 isn't perfect, and it probably never will be unless it was rewritten natively in angular (something I've considered).

**Pull request closed without merging**